### PR TITLE
Tests/combined

### PR DESCRIPTION
**Pull request by YiFeiZhang2**
Create test data and relevant functions for automated test setup.
Create unit tests for current routes of /api/account and /api/hacker

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const indexRouter = require("./routes/index");
 const accountRouter = require("./routes/api/account");
 const authRouter = require("./routes/api/auth");
 const hackerRouter = require("./routes/api/hacker");
-const teamRouter = require("./routes/api/team");
+// const teamRouter = require("./routes/api/team");
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply app.js]: There's nothing in the route, so later 
```javascript
teamRouter.activate(apiRouter);
``` 
gives an error. So I commented that line and this teamRouter definition.

File: controllers/account.controller.js [hunk 2]
```diff
@@ -15,8 +16,8 @@ module.exports = {
     },
 
     // untested
-    getUserByEmail: function (req, res) {
-        const acc = Services.Account.findByEmail(req.user.email);
+    getUserByEmail: Util.asyncMiddleware(async(req, res, next) => {
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply controllers/account.controller.js]: Done

File: controllers/account.controller.js [hunk 3]
```diff
@@ -30,36 +31,35 @@ module.exports = {
...
                 message: "Issue with account creation",
                 data: {}
             });
         }
-    },
+    }),
 
     // untested
-    changeUserInfo: function (req, res) {
+    changeUserInfo: Util.asyncMiddleware(async(req, res, next) => {
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply controllers/account.controller.js]: Done

File: services/account.service.js [hunk 4]
```diff
@@ -24,7 +24,8 @@ async function findByEmail(email) {
     const query = {
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply services/account.service.js]: Forgot to put await. Fixed

File: services/account.service.js [hunk 5]
```diff
@@ -44,35 +45,38 @@ async function getAccountIfValid(email, password) {
 }
 
 function hashPassword(password) {
-    return bcrypt.hashSync(req.body.password, 10);
+    return bcrypt.hashSync(password, 10);
 }
 
 async function findOne(query) {
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply services/account.service.js]: Forgot to re-add await after removing when testing .then. Fixed

File: routes/api/hacker.js [hunk 6]
```diff
@@ -23,16 +23,16 @@ module.exports = {
         const hackerRouter = new express.Router();
 
         // gets a hacker by their email and changes their status
-        hackerRouter.route("/adminChangeHacker").post(
+        hackerRouter.route("/adminChangeHacker/:id").post(
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply routes/api/hacker.js]: I'll do the full found of route changes in the next PR.

File: tests/account.test.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+});
+
+describe("POST create account", function () {
+    it("should SUCCEED and create a new user", function(done) {
+        chai.request(server.app)
+        .post(`/api/account/create`)
+        .type("application/json")
+        .send(newAccount1)
+        .end(function (err, res) {
+            // auth???
```
[the remark this answers is not in the document]
Comment (YiFeiZhang2) [reply tests/account.test.js]: Removed comment.

**Comment by pierreTklein** [comment]
Looks good to me

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**